src/MainWindow.vala

```diff
@@ -620,12 +620,21 @@ namespace PantheonCalculator {
 
         private void show_history (Gtk.Button button) {
             position = entry.get_position ();
-            button_history.sensitive = false;
 
             history_dialog = new HistoryDialog (history);
             history_dialog.set_transient_for (this);
             history_dialog.added.connect (history_added);
-            history_dialog.hide.connect (() => button_history.set_sensitive (true));
+            history_dialog.clear_history.connect (() => {
+                history.foreach ((entry) => {
+                    history.remove_link (history.find (entry));
+                });
```

Comment:
I think `delete_link` should be used if you want to free the entry.  

`history = null` also works.